From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Leandro Dorileo <leandro.maciel.dorileo@intel.com>
Date: Thu, 19 Sep 2019 13:55:12 -0700
Subject: [PATCH] debuginfo: do not strip static libraries

In the future, we will likely not ship static libraries at all, so do not create
debuginfo for them. Likewise, we don't want to strip ld.so for valgrind.

Ported-by: Leandro Dorileo <leandro.maciel.dorileo@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@linux.intel.com>
---
 scripts/find-debuginfo.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/find-debuginfo.sh b/scripts/find-debuginfo.sh
index 87e9511..3c7c61f 100755
--- a/scripts/find-debuginfo.sh
+++ b/scripts/find-debuginfo.sh
@@ -330,9 +330,11 @@ temp=$(mktemp -d ${TMPDIR:-/tmp}/find-debuginfo.XXXXXX)
 trap 'rm -rf "$temp"' EXIT
 
 # Build a list of unstripped ELF files and their hardlinks
+# (also consider non-static libraries)
 touch "$temp/primary"
 find "$RPM_BUILD_ROOT" ! -path "${debugdir}/*.debug" -type f \
      		     \( -perm -0100 -or -perm -0010 -or -perm -0001 \) \
+		     ! -name "*.a" ! -name "ld-*.so" ! -name "ld-linux-x86-64.so*" \
 		     -print |
 file -N -f - | sed -n -e 's/^\(.*\):[ 	]*.*ELF.*, not stripped.*/\1/p' |
 xargs --no-run-if-empty stat -c '%h %D_%i %n' |
